doc/rule-dsl.md

```diff
@@ -0,0 +1,510 @@
+```
+
+#### Property: match.tcp <a id="match-tcp"></a>
+
+TBD
+
+#### Property: match.udp <a id="match-udp"></a>
+
+TBD
+
+#### Property: route <a id="route"></a>
+
+The `route` field identifies a set of one or more the weighted backends to call when the rule is activated.
+Each `route` backend is an object with the following fields:
+
+* `tags`
+* `weight`
+* `destination`
+
+Of these fields, `tags` is the only required one, the others are optional.
+
+#### Property: route.tags <a id="route-tags"></a>
+
+The `tags` field is a list of instance tags to identify the target instances (e.g., version) to route requests to.
+If there are multiple registered instances with the specified tag(s), they will be routed to in a round-robin fashion.
+
+#### Property: route.weight <a id="route-weight"></a>
+
+The `weight` field is an optional value between 0 and 100 that represents the percentage of requests to route
+to instances associated with the corresponding backend. If not set, the `weight` is assumed to be 100.
+
+The sum of all route weights in a rule must equal 100.
+For example, the following rule will route 25% of traffic for the "reviews" service to instances with
+the "v2" tag and the remaining traffic (i.e., 75%) to "v1".
+
+```yaml
+destination: reviews.default.svc.cluster.local
+route:
+- tags:
+    version: v2
+  weight: 25
+- tags:
+    version: v1
+  weight: 75
+```
+
+#### Property: route.destination <a id="route-destination"></a>
+
+The `destination` field is optional and specifies the service name of the target instances. 
+If not specified, it defaults to the value of the rule's `destination` field.
+
+#### Property: http_req_timeout <a id="http_req_timeout"></a>
+
+A timeout for http requests can be specified using the `http_req_timeout` field.
+By default, the timeout is 15 seconds, but this can be overridden as follows:
+
+```yaml
+destination: "ratings.default.svc.cluster.local"
+route:
+- tags:
+    version: v1
+http_req_timeout:
+  simple_timeout:
+    timeout_seconds: 10
+```
+
+Alternatively, you can specify a timeout override header that downstream clients can use to
+dynamically override the timeout on a per request basis:
+
+```yaml
+destination: "ratings.default.svc.cluster.local"
+route:
+- tags:
+    version: v1
+http_req_timeout:
+  simple_timeout:
+    override_header_name: My-Request-Timeout
+```
+
+#### Property: http_req_retries <a id="http_req_retries"></a>
+
+The `http_req_retries` field can be used to control the number retries for a given http request.
+The maximum number of attempts, or as many as possible within the time period
+specified by `http_req_timeout`, can be set as follows:
+
+```yaml
+destination: "ratings.default.svc.cluster.local"
+route:
+- tags:
+    version: v1
+http_req_retries:
+  simple_retry:
+    attempts: 3
+```
+
+Just like timeouts, a retry override header can be specified to allow downstream clients to
+dynamically override the value on a per request basis:
+
```

Comment:
Please remove this as well, or alternatively, we can keep this text and link out to Envoy docs, that specify what the override header name is and the format of the value
https://lyft.github.io/envoy/docs/configuration/http_filters/router_filter.html#x-envoy-upstream-rq-timeout-ms

Either way, we cannot specify our own override header name with Envoy (it was included for the sake of other proxy implementations).